moveit_core/robot_state/test/robot_state_benchmark.cpp

```diff
@@ -37,69 +37,124 @@
 #include <moveit/robot_model/robot_model.h>
 #include <moveit/robot_state/robot_state.h>
 #include <moveit/utils/robot_model_test_utils.h>
+#include <eigen_stl_containers/eigen_stl_containers.h>
 #include <gtest/gtest.h>
-#include <ctime>
+#include <chrono>
 
-class PR2 : public testing::Test
+// Helper class to measure time within a scoped block and output the result
+class ScopedTimer
+{
+  const char* const msg_;
+  double* const gold_standard_;
+  const std::chrono::time_point<std::chrono::steady_clock> start_;
+
+public:
+  // if gold_standard is provided, a relative increase/decrease is shown too
+  ScopedTimer(const char* msg = "", double* gold_standard = nullptr)
+    : msg_(msg), gold_standard_(gold_standard), start_(std::chrono::steady_clock::now())
+  {
+  }
+
+  ~ScopedTimer()
+  {
+    std::chrono::duration<double> elapsed = std::chrono::steady_clock::now() - start_;
+    std::cerr << msg_ << elapsed.count() * 1000. << "ms ";
+
+    if (gold_standard_)
+    {
+      if (*gold_standard_ == 0)
+        *gold_standard_ = elapsed.count();
+      std::cerr << 100 * elapsed.count() / *gold_standard_ << "%";
+    }
+    std::cerr << std::endl;
+  }
+};
+
+class Timing : public testing::Test
 {
 protected:
   void SetUp() override
   {
-    model_ = moveit::core::loadTestingRobotModel("pr2_description");
-  };
+    Eigen::Isometry3d iso = Eigen::Translation3d(1, 2, 3) * Eigen::AngleAxisd(0.13 * M_PI, Eigen::Vector3d::UnitX()) *
+                            Eigen::AngleAxisd(0.29 * M_PI, Eigen::Vector3d::UnitY()) *
+                            Eigen::AngleAxisd(0.42 * M_PI, Eigen::Vector3d::UnitZ());
+    trafos.push_back(Eigen::Isometry3d::Identity());  // result
+    trafos.push_back(iso);                            // input
+  }
 
   void TearDown() override
   {
   }
 
-  robot_model::RobotModelPtr model_;
+public:
+  const Eigen::Isometry3d id = Eigen::Isometry3d::Identity();
+  // put transforms into a vector to avoid compiler optimization on variables
+  EigenSTL::vector_Isometry3d trafos;
+  volatile size_t res = 0;
+  volatile size_t inp = 1;
```

Comment:
How about readable variable names? 
`transforms`, `result_idx` and `input_idx`?

Reply:
Very in favor